pytorch_lightning/core/lightning.py

```diff
@@ -1376,7 +1376,7 @@ def toggle_optimizer(self, optimizer: Union[Optimizer, LightningOptimizer], opti
         # Iterate over all optimizer parameters to preserve their `requires_grad` information
         # in case these are pre-defined during `configure_optimizers`
         param_requires_grad_state = {}
-        for opt in self.optimizers(use_pl_optimizer=False):
```

Comment:
@carmocca (in a separate issue/PR) should we deprecate the `use_pl_optimizers` flag from this method? ideally users would not be exposed to the lightning optimizer at all

Reply:
By default, the users are always exposed the LightningOptimizers. This enables to ensure PL precision and distribution works properly.

Reply:
I think Ananth is saying that users should not need to know about the existence of this wrapper, even if their code actually uses it.

I believe this flag was added so manual opt users had a easy mechanism to get their original optimizers in case the wrapper had an issue. I haven't seen anybody use this flag and we wouldn't want users to need it in general.

IMO we could get rid of it.

Reply:
but what if users want to control the precision grad scalar by themselves?

Reply:
Then they should be using their own precision plugin